### ISSUE TITLE
Concurrent /v1 generation requests corrupt shared decode state (garbled output, non-deterministic greedy, intermittent slice panic)

### ISSUE THREAD

**Issue by VictorAnathorn** [timtoole02/Camelid#302]
## Summary

When two or more generation requests (`/v1/chat/completions`, `/v1/completions`, `/completion`) are in flight at the same time against a loaded model on the CUDA-resident Q8 runtime, they corrupt each other's decode state. Symptoms:

- garbled "word-salad" output (sometimes mixed-script tokens),
- greedy decoding (`temperature: 0`) becoming **non-deterministic**, and
- an intermittent panic in the worker:
  `range end index 7340160 out of range for slice of length 7340032` (the delta is 128 = one attention head_dim).

Serving requests **one at a time** is clean and deterministic — even long (1024-token) generations. The corruption only appears under concurrency / rapid back-to-back requests.

## Why this happens

Generation reaches the GPU-resident weights and KV/decode state through shared `Arc`s under **read** locks (e.g. `loaded_models.read().await`, `cached_weights.read().await`, the shared `cached_prompt_prefix`), and there is no exclusive lock or queue serializing the decode itself. Multiple concurrent decodes therefore run against the same mutable GPU-resident state and clobber each other.

This is consistent with the engine's own reporting: `/props` advertises `total_slots: 1`, and the Q8 support level is `…smoke…` (single-shot evidence), so concurrency was never covered.

## Reproduction

1. `camelid serve --model <a-small-q8_0.gguf>` (CUDA build).
2. Fire 4–6 **identical** requests concurrently with `temperature: 0` (greedy):
   ```json
   POST /v1/chat/completions
   {"model":"<id>","messages":[{"role":"user","content":"List three colors."}],"temperature":0,"max_tokens":24}
   ```
3. Observe:
   - outputs differ across the concurrent batch (greedy over identical inputs must be identical),
   - some replies contain garbled / mixed-script tokens,
   - bursts occasionally return `503` with the out-of-bounds slice panic above.

Greedy non-determinism is the smoking gun: argmax over identical logits cannot differ run-to-run unless the underlying state is being corrupted.

## Expected

Concurrent requests should either be serialized (one decode at a time) or given fully isolated per-request decode state, so output is always coherent and greedy decoding stays deterministic.

## Environment

- Camelid CUDA build (`selected_backend: cuda_resident_q8_runtime`, `cuda_resident_active: true`), Windows x86_64, small Qwen3 Q8_0 model.

## Fix

A pull request follows that serializes generation behind a single in-flight lock (held across the full decode, including the SSE stream), which matches the single-slot model the server already advertises and mirrors what the built-in UI does implicitly.
